cmd/mountlib/mount.go

```diff
@@ -51,6 +65,111 @@ func init() {
 	}
 }
 
+// AddRc adds mount and unmount functionality to rc
+func AddRc(mountFunction MountFn) {
+	// rcMount allows the mount command to be run from rc
+
+	currentMountFn = mountFunction
+
+	rc.Add(rc.Call{
+		Path:         "mount/mount",
+		AuthRequired: true,
+		Fn:           mountRc,
+		Title:        "Create a new mount point",
+		Help: `
+rclone allows Linux, FreeBSD, macOS and Windows to
+mount any of Rclone's cloud storage systems as a file system with
+FUSE.
+
+This takes the following parameters
+
+	- mountPoint: valid path on the local machine (required)
+	- remotePath - a remote path to be mounted (required)
+
+Eg
+
+    rclone rc mount/mount mountPoint=/home/<user>/mountPoint remotePath=mydrive:
+		`,
+	})
+
+	rc.Add(rc.Call{
+		Path:         "mount/unmount",
+		AuthRequired: true,
+		Fn:           unMountRc,
+		Title:        "Unmount all active mounts",
+		Help: `
+rclone allows Linux, FreeBSD, macOS and Windows to
+mount any of Rclone's cloud storage systems as a file system with
+FUSE.
+
+This takes the following parameters
+
+	- mountPoint: valid path on the local machine (required)
+	- remotePath - a remote path to be mounted (required)
+
+Eg
+
+    rclone rc mount/unmount mountPoint=/home/<user>/mountPoint remotePath=mydrive:
+		`,
+	})
+}
+
+// rcMount allows the umount command to be run from rc
+func unMountRc(_ context.Context, _ rc.Params) (out rc.Params, err error) {
+
+	out = make(rc.Params)
+	if currentUnmountFn != nil {
+		err := currentUnmountFn()
+		if err != nil {
+			out["status"] = "Failed"
+			out["err"] = err
+		} else {
+			out["status"] = "Success"
+		}
+	} else {
+		out["status"] = "No mount found"
+	}
+
+	return out, nil
+
+}
+
+// rcMount allows the mount command to be run from rc
+func mountRc(_ context.Context, in rc.Params) (out rc.Params, err error) {
+
+	mountpoint, err := in.GetString("mountPoint")
+	if err != nil {
+		return nil, err
+	}
+	remotePath, err := in.GetString("remotePath")
+	if err != nil {
+		return nil, err
+	}
+
+	out = make(rc.Params)
+
+	tempArgs := []string{remotePath}
+	//append(tempArgs, remotePath)
+	fdst := cmd.NewFsDir(tempArgs)
+
+	_, _, currentUnmountFn, err = currentMountFn(fdst, mountpoint)
```

Comment:
So we should be stashing the unmountfn in a dictionary keyed off mountpoint probably...